enhancements/node/cri-o-operator.md

```diff
@@ -0,0 +1,307 @@
+
+## Proposal
+
+The main proposal is integrating a new operator into OpenShift, which is owned
+by the Node team. This includes the whole supply chain around CI/CD and release
+management.
+
+The operator will consist of:
+
+- A dedicated namespace where the operator lives and acts on.
+- A controller managing the reconciliation of the statically provided assets,
+  which are:
+  - RBAC rules as well as a ServiceAccount for the kube-rbac-proxy based metrics
+    server.
+  - SecurityContextConstraints to allow HostPorts and HostNetwork for the
+    DaemonSet.
+  - A DaemonSet to retrieve the metrics for every node.
+  - A secret for the TLS cert/key for the metrics retrieval (used by CRI-O as
+    well as the kube-rbac-proxy).
+  - A service to access the metrics.
+  - A ServiceMonitor to integrate the metrics into OpenShift.
+- A reconciler for the required TLS cert/key. This reconciliation loop will
+  recreate the certificate if nearly expired and will sync them to disk (for
+  CRI-O) as well as into the existing secret (for kube-rbac-proxy). CRI-O
+  already supports on-the-fly certificate reload for its secure metrics
+  endpoint.
+
+A proof of concept integration can be found [within the CMO][poc].
+
+Because the whole monitoring stack should be integrated into the OpenShift
+cluster monitoring (not the user workload monitoring), it may be possible that
+the CMO has to be modified to accept the ServiceMonitor within the CRI-O
+operator namepsace. A corresponing RBAC cluster role binding to allow the
+metrics retrieval between both namespaces will be required, too:
+
+```yaml
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: crio-metrics-client
+rules:
+  - nonResourceURLs:
+      - /metrics
+    verbs:
+      - get
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: crio-metrics-client
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: crio-metrics-client
+subjects:
+  - kind: ServiceAccount
+    name: prometheus-k8s
+    namespace: openshift-monitoring
+```
+
+[poc]: https://github.com/openshift/cluster-monitoring-operator/pull/1219
+
+### User Stories
+
+As a Node team, I want to have full control over how CRI-O metrics get
+deployed.
+
+As a Node team, I want to ensure future proofness for the latest development
+efforts in OpenShift.
+
+As a cluster operator, I want to ensure that CRI-O metrics endpoints are secure
+by default and covered by RBAC rules.
+
+### Risks and Mitigations
+
+#### A new project to integrate
+
+The highest risk for OpenShift as well as the Node team is that we have to
+maintain and integrate a new operator into the product. This will add initially
+higher efforts on the team, while gaining more flexibility in a mid-to-long
+term. A mitigation around this to keep the focus on metrics in the first place,
+while having in mind to change the operator later on to fulfill the APIs from
+the MCO and CMO.
```

Comment:
maybe also mention the risk of having a new operator managed by CVO (if we want that--it seems like we may)

Reply:
Added the risk :+1: